geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/ClientServerFunctionExecutionDUnitTest.java

```diff
@@ -430,6 +432,42 @@ public void testOnServerExecution_FunctionInvocationTargetException() {
             Boolean.FALSE));
   }
 
+  @Test
+  public void onRegionShouldThrowExceptionWhenThePoolAssociatedWithTheRegionCanNotBeFound() {
+    function = new TestFunction(true, TEST_FUNCTION1);
+    createScenario();
+    registerFunctionAtServer(function);
+
+    server1.invoke(ClientServerFunctionExecutionDUnitTest::createReplicatedRegion);
+    server2.invoke(ClientServerFunctionExecutionDUnitTest::createReplicatedRegion);
+    server3.invoke(ClientServerFunctionExecutionDUnitTest::createReplicatedRegion);
+
+    client.invoke(() -> {
+      ClientServerFunctionExecutionDUnitTest
+          .createProxyRegion(NetworkUtils.getServerHostName(server1.getHost()));
+      assertThatThrownBy(() -> HijackedFunctionService.onRegion(metaDataRegion).execute(function))
+          .isInstanceOf(IllegalStateException.class)
+          .hasMessageMatching("Could not find a pool named (.*)");
+    });
+  }
+
+  private static class HijackedFunctionService extends FunctionService {
+    public HijackedFunctionService(FunctionExecutionService functionExecutionService) {
+      super(functionExecutionService);
+    }
+
+    public static Execution onRegion(Region region) {
+      return new HijackedInternalFunctionServiceImpl().onRegion(region);
```

Comment:
Just leaving this here to reinforce my understanding - the reason we need to stub different behavior on the FunctionService in this test is because your fixes prevent us from getting into the state originally described in the ticket right?  (Pool cannot be removed if it is still in use).

Reply:
Yes, that's exactly right.
Also, and just FYI, I chose to stub the behaviour instead of mocking `FunctionService` because all methods are static, so I would have had to use `PowerMock` or something similar, discouraged according to my understanding.
Cheers.

Reply:
Yep that makes sense.  I think not using PowerMock is the right call, and I don't take issue with your testing approach.  It was more trying to understand why we needed to stub at all, and the answer is that your fixes have made it impossible to get into the original problematic state as described by the ticket.  Thanks for answering my questions!